code/extension/Member.php

```diff
@@ -88,6 +90,12 @@ public function OTPSeed()
      */
     public function updateCMSFields(\FieldList $fields)
     {
+        // Generate default token (allows scanning the QR at the moment of activation and (optionally) validate before activating 2FA)
+        if(!$this->owner->TOTPToken && !\Config::inst()->get(__CLASS__, 'regenerate_on_activation')) {
```

Comment:
@oddnoc happy with this implementation? Wondering if we start thinking about some unit tests as the complexity starts to grow and there are more configurable options.